executor/cleaner/cleaner_test.go

```diff
@@ -3,37 +3,31 @@
 package cleaner_test
 
 import (
-	"context"
 	"testing"
 
 	k8upv1a1 "github.com/vshn/k8up/api/v1alpha1"
+	"github.com/vshn/k8up/envtest"
 	"github.com/vshn/k8up/executor/cleaner"
 	"github.com/vshn/k8up/job"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
-	"github.com/go-logr/logr"
-	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/uuid"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
-func TestCleanup(t *testing.T) {
-	jobs := jobList(3, 2, 2)
-	client := newMockClient(jobs)
+type CleanerTestSuite struct {
+	envtest.Suite
+}
 
-	objCleaner := &cleaner.ObjectCleaner{Client: client, Limits: newLimiter(1, 1), Log: logr.DiscardLogger{}}
-	deleted, err := objCleaner.CleanOldObjects(context.TODO(), jobs.GetJobObjects())
-	assert.NoError(t, err)
-	assert.Equal(t, 2, deleted)
+func Test_Cleaner(t *testing.T) {
+	suite.Run(t, new(CleanerTestSuite))
+}
 
-	afterClean := &k8upv1a1.RestoreList{}
-	assert.NoError(t, client.List(context.TODO(), afterClean))
-	runningJobs, failedJobs, successfulJobs := job.GroupByStatus(afterClean.GetJobObjects())
-	assert.Equal(t, 3, len(runningJobs))
-	assert.Equal(t, 1, len(failedJobs))
-	assert.Equal(t, 1, len(successfulJobs))
+func (ts *CleanerTestSuite) TestCleanup() {
+	ts.withJobs()
+	ts.runCleanup()
+	ts.checkIfJobsDeleted()
 }
```

Comment:
Nice! Clean and readable. Just a small suggestion
```suggestion
func (ts *CleanerTestSuite) TestCleanup() {
	ts.withJobs()
	ts.runCleanup()
	ts.assertJobsDeleted()
}
```

Reply:
sounds better 👍